features/frameworks/nanostack-libservice/mbed-client-libservice/ns_types.h

```diff
@@ -120,7 +120,9 @@ typedef int_fast32_t int_fast24_t;
 #if defined __CC_ARM || defined __TASKING__
 #define alignas(n) __align(n)
 #define __alignas_is_defined 1
-#elif (defined __STDC_VERSION__ && __STDC_VERSION__ >= 201112L) || (defined __cplusplus && __cplusplus >= 201103L)
+#elif defined __cplusplus && __cplusplus >= 201103L
+#define __alignas_is_defined 1
+#elif defined __STDC_VERSION__ && __STDC_VERSION__ >= 201112L
```

Comment:
This header is included mainly by C files - doesn't it still trip up? I guess you get away with it because you haven't switched ARMC6 to C11 mode yet. We will be doing that.

I'd rather have a solution that makes ARMC6 work fully - if we can't fix ARMC6, then detect it specifically and provide the stdalign.h substitute for both C and C++.

I'd be somewhat inclined to put that in mbed_retarget.h or something, as there will be other places wanting to use alignas.

Reply:
Although actually this is a separate portability glue used outside Mbed OS too (mainly for C) so something still needs to be here.